lib/Target/LLVMIR/LLVMIRTranslation.cpp

```diff
@@ -145,10 +148,47 @@ translateTritonGPUToLLVMIR(llvm::LLVMContext *llvmContext,
     llvm::errs() << "Pass execution failed";
     return nullptr;
   }
+  SmallVector<LLVM::LLVMFuncOp> funcs;
+  module.walk([&](LLVM::LLVMFuncOp func){
+    funcs.push_back(func);
+  });
+
+  static const std::string key("libpath:");
+  std::set<std::string> extern_libs;
+  for (auto &func : funcs) {
+    if (!func.isExternal())
+      continue;
+    auto attrs = func.getPassthroughAttr();
+    for (auto& attr : attrs) {
+      if (StringAttr sAttr = attr.dyn_cast<StringAttr>()) {
+        auto value = sAttr.str();
+        if (0 == value.find(key)) {
+          extern_libs.insert(value.substr(key.size()));
+        }
+      }
+    }
+  }
 
   auto llvmir = translateLLVMToLLVMIR(llvmContext, module);
   if (!llvmir) {
     llvm::errs() << "Translate to LLVM IR failed";
+    return nullptr;
+  }
+
+  llvm::SMDiagnostic err;
+  for (auto& path : extern_libs) {
+    auto ext_mod = llvm::parseIRFile(path, err, *llvmContext);
```

Comment:
maybe we should document some TODO here. parseIRFile() could be done in a static way in case we are sensitive to the compilation overhead.

Reply:
Here it need more investigation,  because llvm::module parsing depends on llvmContext, so it looks it is one time thing.